### PR TITLE
Move `declare` and `deploy` command docs to "Interacting with StarkNet"

### LINKED ISSUE #1099: Add `declare` and `deploy` to "Interacting with StarkNet" docs
Move from 
Deployment > CLI
to 
Interacting with StarkNet

WDYT @kasperski95 @mkaput ?

### PR DESCRIPTION
**Pull request by Arcticae**
closes #1099

### CODE REVIEW

File: website/docs/tutorials/09-interacting-with-starknet/03-declare.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# Declaring new contracts
```
Comment (mkaput) [inline website/docs/tutorials/09-interacting-with-starknet/03-declare.md:1]: 1. Where is `05-deploy-account`?
2. There are two `09-*` pages: interacting with starknet & migrations

**Review by mkaput — CHANGES_REQUESTED** [review]
Consider applying tweaks from #1116

File: website/docs/tutorials/09-interacting-with-starknet/03-declare.md (hunk 1 above)
Comment (Arcticae) [reply website/docs/tutorials/09-interacting-with-starknet/03-declare.md]: 1. i moved it back to 05
2. done

File: website/docs/tutorials/07-testing/02-cheatcodes/deploy-contract.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ class DeployedContract:
 Deploys a contract given a path relative to a Protostar project root. The section [Deploying contracts from tests](../01-deploying-contracts.md) demonstrates a usage of this cheatcode.
 
 - `contract` — A path to the contract's main `.cairo` source file.
-- `config` — A keyword only argument kept for compatibility with the migration [deploy_contract cheatcode](../../08-deploying/02-migrations/deploy-contract.md). See related documentation for more information.
+- `config` — A keyword only argument kept for compatibility with the migration [deploy_contract cheatcode](../../09-migrations/deploy-contract.md). See related documentation for more information.
```
[the remark this answers is not in the document]
Comment (Arcticae) [reply website/docs/tutorials/07-testing/02-cheatcodes/deploy-contract.md:17]: i added a redirect

**Review by mkaput — APPROVED** [review]

**Pull request merged**